h2o-r/h2o-package/R/explain.R

```diff
@@ -3394,7 +3412,14 @@ h2o.explain_row <- function(object,
       columns_of_interest <- sapply(columns_of_interest, .find_appropriate_column_name, cols = models_info$x)
     }
   }
-
+  # Make sure that there are no string columns to explain as they are not supported by pdp
+  # Usually those columns would not be used by algos so this just makes sure to exclude them
+  # if user specifies top_n = Inf or columns_of_interest = x etc.
+  dropped_string_columns <- Filter(function(col) h2o.getTypes(newdata[[col]])[[1]] == "string", columns_of_interest)
+  if (length(dropped_string_columns) > 1) {
+    warning(sprintf("Dropping string columns as they are unsupported: %s", paste(dropped_string_columns, collapse = ", ")), call. = FALSE)
+    columns_of_interest <- Filter(function(col) h2o.getTypes(newdata[[col]])[[1]] != "string", columns_of_interest)
```

Comment:
I guess efficient partitioning is also doable in R, but after scratching my head, doesn't look that simple :)